<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Structure and content of RAVmodel • GenomicSuperignature</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Structure and content of RAVmodel">
<meta property="og:description" content="GenomicSuperSignature">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">GenomicSuperignature</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Quickstart.html">Quick Start Guide</a>
    </li>
    <li>
      <a href="../articles/GenomicSuperSignature_Contents.html">Model Contents</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/shbrief/GenomicSuperSignature">
    <span class="fas fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="GenomicSuperSignature_Contents_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Structure and content of RAVmodel</h1>
                        <h4 class="author">Sehyun Oh</h4>
            
            <h4 class="date">May 24, 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/shbrief/GenomicSuperSignature/blob/master/vignettes/GenomicSuperSignature_Contents.Rmd"><code>vignettes/GenomicSuperSignature_Contents.Rmd</code></a></small>
      <div class="hidden name"><code>GenomicSuperSignature_Contents.Rmd</code></div>

    </div>

    
    
<div id="setup" class="section level1">
<h1 class="hasAnchor">
<a href="#setup" class="anchor"></a>Setup</h1>
<div id="install-and-load-package" class="section level2">
<h2 class="hasAnchor">
<a href="#install-and-load-package" class="anchor"></a>Install and load package</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"GenomicSuperSignature"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/shbrief/GenomicSuperSignature">GenomicSuperSignature</a></span><span class="op">)</span></code></pre></div>
</div>
<div id="download-ravmodel" class="section level2">
<h2 class="hasAnchor">
<a href="#download-ravmodel" class="anchor"></a>Download RAVmodel</h2>
<p>You can download GenomicSuperSignature from Google Cloud bucket using <code><a href="../reference/getModel.html">GenomicSuperSignature::getModel</a></code> function. Currently available models are built from top 20 PCs of 536 studies (containing 44,890 samples) containing 13,934 common genes from each of 536 study’s top 90% varying genes based on their study-level standard deviation. There are two versions of this RAVmodel annotated with different gene sets for GSEA: MSigDB C2 (<code>C2</code>) and three priors from PLIER package (<code>PLIERpriors</code>). In this vignette, we are showing the <code>C2</code> annotated model.</p>
<p>Note that the first interactive run of this code, you will be asked to allow R to create a cache directory. The model file will be stored there and subsequent calls to <code>getModel</code> will read from the cache.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">RAVmodel</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getModel.html">getModel</a></span><span class="op">(</span><span class="st">"C2"</span>, load<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#&gt; [1] "downloading"</span></code></pre></div>
</div>
</div>
<div id="content-of-ravmodel" class="section level1">
<h1 class="hasAnchor">
<a href="#content-of-ravmodel" class="anchor"></a>Content of RAVmodel</h1>
<p><code>RAVindex</code> is a matrix containing genes in rows and RAVs in columns. <code>colData</code> slot provides the information on each RAVs, such as GSEA annotation and studies involved in each cluster. <code>metadata</code> slot stores model construction information. <code>trainingData</code> slot contains the information on individual studies in training dataset, such as MeSH terms assigned to each study.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">RAVmodel</span>
<span class="co">#&gt; class: PCAGenomicSignatures </span>
<span class="co">#&gt; dim: 13934 4764 </span>
<span class="co">#&gt; metadata(8): cluster size ... updateNote version</span>
<span class="co">#&gt; assays(1): RAVindex</span>
<span class="co">#&gt; rownames(13934): CASKIN1 DDX3Y ... CTC-457E21.9 AC007966.1</span>
<span class="co">#&gt; rowData names(0):</span>
<span class="co">#&gt; colnames(4764): RAV1 RAV2 ... RAV4763 RAV4764</span>
<span class="co">#&gt; colData names(4): RAV studies silhouetteWidth gsea</span>
<span class="co">#&gt; trainingData(2): PCAsummary MeSH</span>
<span class="co">#&gt; trainingData names(536): DRP000987 SRP059172 ... SRP164913 SRP188526</span></code></pre></div>
<center>
<img src="https://raw.githubusercontent.com/shbrief/GenomicSuperSignaturePaper/master/inst/images/GSig_RAVmodel.png" width="90%" height="90%">
</center>
<div id="ravindex" class="section level2">
<h2 class="hasAnchor">
<a href="#ravindex" class="anchor"></a>RAVindex</h2>
<p><em>R</em>eplicable <em>A</em>xis of <em>V</em>ariation (RAV) index is the main component of GenomicSuperSignature. It serves as an index connecting new datasets and the existing database. You can access it through <code><a href="../reference/GenomicSignatures-methods.html">GenomicSuperSignature::RAVindex</a></code> (equivalent of <code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">SummarizedExperiment::assay</a></code>). Rows are genes and columns are RAVs.</p>
<p>Here, RAVmodel consists of 13,934 genes and 4,764 RAVs.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="fu"><a href="../reference/GenomicSignatures-methods.html">RAVindex</a></span><span class="op">(</span><span class="va">RAVmodel</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] "matrix" "array"</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="fu"><a href="../reference/GenomicSignatures-methods.html">RAVindex</a></span><span class="op">(</span><span class="va">RAVmodel</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] 13934  4764</span>
<span class="fu"><a href="../reference/GenomicSignatures-methods.html">RAVindex</a></span><span class="op">(</span><span class="va">RAVmodel</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>
<span class="co">#&gt;                RAV1          RAV2         RAV3         RAV4</span>
<span class="co">#&gt; CASKIN1 0.002449532 -2.845471e-03  0.001601715 -0.009366156</span>
<span class="co">#&gt; DDX3Y   0.006093438  1.870528e-03 -0.004346019 -0.006094526</span>
<span class="co">#&gt; MDGA1   0.009999975  2.610302e-03 -0.002831447 -0.006710757</span>
<span class="co">#&gt; EIF1AY  0.007923652  8.507021e-05  0.002322275 -0.007850784</span></code></pre></div>
</div>
<div id="metadata-for-ravmodel" class="section level2">
<h2 class="hasAnchor">
<a href="#metadata-for-ravmodel" class="anchor"></a>Metadata for RAVmodel</h2>
<p>Metadata slot of RAVmodel contains information related to the model building.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="fu"><a href="../reference/GenomicSignatures-methods.html">metadata</a></span><span class="op">(</span><span class="va">RAVmodel</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] "cluster"    "size"       "k"          "n"          "geneSets"  </span>
<span class="co">#&gt; [6] "MeSH_freq"  "updateNote" "version"</span></code></pre></div>
<ul>
<li>
<code>cluster</code> : cluster membership of each PCs from the training dataset<br>
</li>
<li>
<code>size</code> : an integer vector with the length of clusters, containing the number of PCs in each cluster<br>
</li>
<li>
<code>k</code> : the number of all clusters in the given RAVmodel<br>
</li>
<li>
<code>n</code> : the number of top PCs kept from each study in the training dataset</li>
<li>
<code>geneSets</code> : the name of gene sets used for GSEA annotation<br>
</li>
<li>
<code>MeSH_freq</code> : the frequency of MeSH terms associated with the training dataset. MeSH terms like ‘Humans’ and ‘RNA-seq’ are top ranked (which is very expected) because the training dataset of this model is Human RNA sequencing data.</li>
<li>
<code>updateNote</code> : a brief note on the given model’s specification</li>
<li>
<code>version</code> : the version of the given model</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/GenomicSignatures-methods.html">metadata</a></span><span class="op">(</span><span class="va">RAVmodel</span><span class="op">)</span><span class="op">$</span><span class="va">cluster</span><span class="op">)</span>
<span class="co">#&gt; DRP000987.PC1 DRP000987.PC2 DRP000987.PC3 DRP000987.PC4 DRP000987.PC5 </span>
<span class="co">#&gt;             1             2             3             4             5 </span>
<span class="co">#&gt; DRP000987.PC6 </span>
<span class="co">#&gt;             6</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/GenomicSignatures-methods.html">metadata</a></span><span class="op">(</span><span class="va">RAVmodel</span><span class="op">)</span><span class="op">$</span><span class="va">size</span><span class="op">)</span>
<span class="co">#&gt; RAV1 RAV2 RAV3 RAV4 RAV5 RAV6 </span>
<span class="co">#&gt;    6   21    4    7    3    3</span>
<span class="fu"><a href="../reference/GenomicSignatures-methods.html">metadata</a></span><span class="op">(</span><span class="va">RAVmodel</span><span class="op">)</span><span class="op">$</span><span class="va">k</span>
<span class="co">#&gt; [1] 4764</span>
<span class="fu"><a href="../reference/GenomicSignatures-methods.html">metadata</a></span><span class="op">(</span><span class="va">RAVmodel</span><span class="op">)</span><span class="op">$</span><span class="va">n</span>
<span class="co">#&gt; [1] 20</span>
<span class="fu"><a href="../reference/GenomicSignatures-methods.html">geneSets</a></span><span class="op">(</span><span class="va">RAVmodel</span><span class="op">)</span>
<span class="co">#&gt; [1] "MSigDB C2 v.7.1"</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/GenomicSignatures-methods.html">metadata</a></span><span class="op">(</span><span class="va">RAVmodel</span><span class="op">)</span><span class="op">$</span><span class="va">MeSH_freq</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;                 Humans                RNA-Seq          Transcriptome </span>
<span class="co">#&gt;                   1163                    591                    576 </span>
<span class="co">#&gt;                    RNA Whole Exome Sequencing        Sequencing, RNA </span>
<span class="co">#&gt;                    372                    284                    277</span>
<span class="fu"><a href="../reference/GenomicSignatures-methods.html">updateNote</a></span><span class="op">(</span><span class="va">RAVmodel</span><span class="op">)</span>  
<span class="co">#&gt; [1] "536 refine.bio studies/ top 90% varying genes/ GSEA with MSigDB C2"</span>
<span class="fu"><a href="../reference/GenomicSignatures-methods.html">metadata</a></span><span class="op">(</span><span class="va">RAVmodel</span><span class="op">)</span><span class="op">$</span><span class="va">version</span>
<span class="co">#&gt; [1] "&gt;= 0.0.7"</span></code></pre></div>
</div>
<div id="studies-in-each-rav" class="section level2">
<h2 class="hasAnchor">
<a href="#studies-in-each-rav" class="anchor"></a>Studies in each RAV</h2>
<p>You can find which studies are in each cluster using <code>studies</code> method. Output is a list with the length of clusters, where each element is a character vector containing the name of studies in each cluster.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="../reference/PCAGenomicSignatures-methods.html">studies</a></span><span class="op">(</span><span class="va">RAVmodel</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] 4764</span>
<span class="fu"><a href="../reference/PCAGenomicSignatures-methods.html">studies</a></span><span class="op">(</span><span class="va">RAVmodel</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span>
<span class="co">#&gt; $Cl4764_01</span>
<span class="co">#&gt; [1] "DRP000987" "SRP059172" "SRP059959" "SRP118741" "SRP144178" "SRP183186"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Cl4764_02</span>
<span class="co">#&gt;  [1] "DRP000987" "ERP022839" "ERP104463" "ERP106996" "ERP109789" "ERP114435"</span>
<span class="co">#&gt;  [7] "SRP059039" "SRP059172" "SRP065317" "SRP071547" "SRP073061" "SRP074274"</span>
<span class="co">#&gt; [13] "SRP075449" "SRP092158" "SRP096736" "SRP107337" "SRP111402" "SRP125008"</span>
<span class="co">#&gt; [19] "SRP133442" "SRP135684" "SRP155483"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Cl4764_03</span>
<span class="co">#&gt; [1] "DRP000987" "SRP001540" "SRP056840" "SRP125396"</span></code></pre></div>
<p>You can check which PC from different studies are in RAVs using <code>PCinRAV</code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/PCinRAV.html">PCinRAV</a></span><span class="op">(</span><span class="va">RAVmodel</span>, <span class="fl">2</span><span class="op">)</span>
<span class="co">#&gt;  [1] "DRP000987.PC2" "ERP022839.PC2" "ERP104463.PC1" "ERP106996.PC2"</span>
<span class="co">#&gt;  [5] "ERP109789.PC2" "ERP114435.PC1" "SRP059039.PC3" "SRP059172.PC2"</span>
<span class="co">#&gt;  [9] "SRP065317.PC1" "SRP071547.PC3" "SRP073061.PC2" "SRP074274.PC5"</span>
<span class="co">#&gt; [13] "SRP075449.PC2" "SRP092158.PC2" "SRP096736.PC2" "SRP107337.PC3"</span>
<span class="co">#&gt; [17] "SRP111402.PC1" "SRP125008.PC2" "SRP133442.PC1" "SRP135684.PC4"</span>
<span class="co">#&gt; [21] "SRP155483.PC2"</span></code></pre></div>
</div>
<div id="silhouette-width-for-each-rav" class="section level2">
<h2 class="hasAnchor">
<a href="#silhouette-width-for-each-rav" class="anchor"></a>Silhouette width for each RAV</h2>
<p>Silhouette width ranges from -1 to 1 for each cluster. Typically, it is interpreted as follows:<br>
- Values close to 1 suggest that the observation is well matched to the assigned cluster<br>
- Values close to 0 suggest that the observation is borderline matched between two clusters<br>
- Values close to -1 suggest that the observations may be assigned to the wrong cluster</p>
<p>For RAVmodel, the average silhouette width of each cluster is a quality control measure and suggested as a secondary reference to choose proper RAVs, following validation score.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PCAGenomicSignatures-methods.html">silhouetteWidth</a></span><span class="op">(</span><span class="va">RAVmodel</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>   <span class="co"># average silhouette width of the first 6 RAVs</span>
<span class="co">#&gt;           1           2           3           4           5           6 </span>
<span class="co">#&gt; -0.05470163  0.06426256 -0.01800335 -0.04005584  0.05786189 -0.02520973</span></code></pre></div>
</div>
<div id="gsea-on-each-rav" class="section level2">
<h2 class="hasAnchor">
<a href="#gsea-on-each-rav" class="anchor"></a>GSEA on each RAV</h2>
<p>Pre-processed GSEA results on each RAV are stored in RAVmodel and can be accessed through <code>gsea</code> function.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="fu"><a href="../reference/PCAGenomicSignatures-methods.html">gsea</a></span><span class="op">(</span><span class="va">RAVmodel</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] "list"</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="fu"><a href="../reference/PCAGenomicSignatures-methods.html">gsea</a></span><span class="op">(</span><span class="va">RAVmodel</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="co">#&gt; [1] "data.frame"</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="../reference/PCAGenomicSignatures-methods.html">gsea</a></span><span class="op">(</span><span class="va">RAVmodel</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] 4764</span>
<span class="fu"><a href="../reference/PCAGenomicSignatures-methods.html">gsea</a></span><span class="op">(</span><span class="va">RAVmodel</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>
<span class="co">#&gt; $RAV1</span>
<span class="co">#&gt;                                                                                     Description</span>
<span class="co">#&gt; HELLEBREKERS_SILENCED_DURING_TUMOR_ANGIOGENESIS HELLEBREKERS_SILENCED_DURING_TUMOR_ANGIOGENESIS</span>
<span class="co">#&gt; KIM_GLIS2_TARGETS_UP                                                       KIM_GLIS2_TARGETS_UP</span>
<span class="co">#&gt; NAKAYAMA_SOFT_TISSUE_TUMORS_PCA2_UP                         NAKAYAMA_SOFT_TISSUE_TUMORS_PCA2_UP</span>
<span class="co">#&gt; PHONG_TNF_TARGETS_UP                                                       PHONG_TNF_TARGETS_UP</span>
<span class="co">#&gt; RASHI_RESPONSE_TO_IONIZING_RADIATION_2                   RASHI_RESPONSE_TO_IONIZING_RADIATION_2</span>
<span class="co">#&gt; REACTOME_TRANSLATION                                                       REACTOME_TRANSLATION</span>
<span class="co">#&gt; ZHU_CMV_ALL_DN                                                                   ZHU_CMV_ALL_DN</span>
<span class="co">#&gt; GALINDO_IMMUNE_RESPONSE_TO_ENTEROTOXIN                   GALINDO_IMMUNE_RESPONSE_TO_ENTEROTOXIN</span>
<span class="co">#&gt;                                                       NES       pvalue</span>
<span class="co">#&gt; HELLEBREKERS_SILENCED_DURING_TUMOR_ANGIOGENESIS -3.851302 1.000000e-10</span>
<span class="co">#&gt; KIM_GLIS2_TARGETS_UP                            -3.472416 1.000000e-10</span>
<span class="co">#&gt; NAKAYAMA_SOFT_TISSUE_TUMORS_PCA2_UP             -3.176200 1.000000e-10</span>
<span class="co">#&gt; PHONG_TNF_TARGETS_UP                            -3.276382 1.000000e-10</span>
<span class="co">#&gt; RASHI_RESPONSE_TO_IONIZING_RADIATION_2          -4.196536 1.000000e-10</span>
<span class="co">#&gt; REACTOME_TRANSLATION                             2.300159 1.000000e-10</span>
<span class="co">#&gt; ZHU_CMV_ALL_DN                                  -3.423609 1.000000e-10</span>
<span class="co">#&gt; GALINDO_IMMUNE_RESPONSE_TO_ENTEROTOXIN          -3.032082 1.139605e-10</span>
<span class="co">#&gt;                                                     qvalues</span>
<span class="co">#&gt; HELLEBREKERS_SILENCED_DURING_TUMOR_ANGIOGENESIS 3.43231e-08</span>
<span class="co">#&gt; KIM_GLIS2_TARGETS_UP                            3.43231e-08</span>
<span class="co">#&gt; NAKAYAMA_SOFT_TISSUE_TUMORS_PCA2_UP             3.43231e-08</span>
<span class="co">#&gt; PHONG_TNF_TARGETS_UP                            3.43231e-08</span>
<span class="co">#&gt; RASHI_RESPONSE_TO_IONIZING_RADIATION_2          3.43231e-08</span>
<span class="co">#&gt; REACTOME_TRANSLATION                            3.43231e-08</span>
<span class="co">#&gt; ZHU_CMV_ALL_DN                                  3.43231e-08</span>
<span class="co">#&gt; GALINDO_IMMUNE_RESPONSE_TO_ENTEROTOXIN          3.43231e-08</span></code></pre></div>
</div>
<div id="mesh-terms-for-each-study" class="section level2">
<h2 class="hasAnchor">
<a href="#mesh-terms-for-each-study" class="anchor"></a>MeSH terms for each study</h2>
<p>You can find MeSH terms associated with each study using <code>mesh</code> method. Output is a list with the length of studies used for training. Each element of this output list is a data frame containing the assigned MeSH terms and the detail of them. The last column <code>bagOfWords</code> is the frequency of the MeSH term in the whole training dataset.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="../reference/PCAGenomicSignatures-methods.html">mesh</a></span><span class="op">(</span><span class="va">RAVmodel</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] 536</span>
<span class="fu"><a href="../reference/PCAGenomicSignatures-methods.html">mesh</a></span><span class="op">(</span><span class="va">RAVmodel</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>
<span class="co">#&gt; $DRP000987</span>
<span class="co">#&gt;          score identifier                   name</span>
<span class="co">#&gt; 1468242 162422  DRP000987                Animals</span>
<span class="co">#&gt; 1596889 162422  DRP000987                 Humans</span>
<span class="co">#&gt; 1116527 161422  DRP000987  Plasmodium falciparum</span>
<span class="co">#&gt; 832289   96999  DRP000987    Malaria, Falciparum</span>
<span class="co">#&gt; 216848   74145  DRP000987              Parasites</span>
<span class="co">#&gt; 1689323  57176  DRP000987          Transcriptome</span>
<span class="co">#&gt; 761253   55509  DRP000987                Malaria</span>
<span class="co">#&gt; 149067   37162  DRP000987              Indonesia</span>
<span class="co">#&gt; 1627781  36315  DRP000987 Transcriptome Analysis</span>
<span class="co">#&gt; 716101   22634  DRP000987                RNA-Seq</span>
<span class="co">#&gt; 1573542  19938  DRP000987                    RNA</span>
<span class="co">#&gt; 58627    18575  DRP000987               Genotype</span>
<span class="co">#&gt; 1230230   7197  DRP000987 Innate Immune Response</span>
<span class="co">#&gt; 35822     1000  DRP000987      Genetic Variation</span>
<span class="co">#&gt;                                                                                                                                                                                                                                                                                                  explanation</span>
<span class="co">#&gt; 1468242                                                                                                                                                                                                                                           Parasites, CT Treecode Lookup: B01.050.500.714 (Parasites)</span>
<span class="co">#&gt; 1596889 Forced Leaf Node Lookup:humans, CT Text Lookup: human, CT Text Lookup: humans, CT Text Lookup: patient, CT Text Lookup: patients, Forced Leaf Node Lookup:humans, CT Text Lookup: human, CT Text Lookup: humans, CT Text Lookup: patient, Parasites, CT Treecode Lookup: B01.050.500.714 (Parasites)</span>
<span class="co">#&gt; 1116527                                                                                                                                                                                                 RtM via: Plasmodium falciparum, Forced Leaf Node Lookup:plasmodium falciparums, checkForceMH Boosted</span>
<span class="co">#&gt; 832289                                                                                                                                                                                                                                                              RtM via: Plasmodium falciparum infection</span>
<span class="co">#&gt; 216848                                                                                                                                                                                                                           RtM via: Parasites, Forced Leaf Node Lookup:parasites, checkForceMH Boosted</span>
<span class="co">#&gt; 1689323                                                                                                                                                                                                                                       RtM via: Transcriptome, Forced Leaf Node Lookup:transcriptomes</span>
<span class="co">#&gt; 761253                                                                                                                                                                                                                                                 RtM via: Malaria, Forced Non-Leaf Node Lookup:malaria</span>
<span class="co">#&gt; 149067                                                                                                                                                                                                                         RtM via: Indonesia, RtM via: Indonesians, Forced Leaf Node Lookup:indonesians</span>
<span class="co">#&gt; 1627781                                                                                                                                                       Entry Term Replacement for "Gene Expression Profiling", RtM via: Gene Expression Profiling, Forced Non-Leaf Node Lookup:transcriptome analysis</span>
<span class="co">#&gt; 716101                                                                                                                                                                                                                                                           RtM via: RNA-Seq, Forced New Lookup:rna-seq</span>
<span class="co">#&gt; 1573542                                                                                                                                                                                                                                                                                         RtM via: RNA</span>
<span class="co">#&gt; 58627                                                                                                                                                                                                                                                                               Forced Lookup:genotyping</span>
<span class="co">#&gt; 1230230                                                                                                                                                                        Entry Term Replacement for "Immunity, Innate", RtM via: Immunity, Innate, Forced Non-Leaf Node Lookup:innate immune responses</span>
<span class="co">#&gt; 35822                                                                                                                                                                                                                                                         Forced Non-Leaf Node Lookup:genetic variations</span>
<span class="co">#&gt;         bagOfWords</span>
<span class="co">#&gt; 1468242        209</span>
<span class="co">#&gt; 1596889       1163</span>
<span class="co">#&gt; 1116527          5</span>
<span class="co">#&gt; 832289           4</span>
<span class="co">#&gt; 216848           7</span>
<span class="co">#&gt; 1689323        576</span>
<span class="co">#&gt; 761253           6</span>
<span class="co">#&gt; 149067           2</span>
<span class="co">#&gt; 1627781         66</span>
<span class="co">#&gt; 716101         591</span>
<span class="co">#&gt; 1573542        372</span>
<span class="co">#&gt; 58627           20</span>
<span class="co">#&gt; 1230230          8</span>
<span class="co">#&gt; 35822           11</span></code></pre></div>
</div>
<div id="pca-summary-for-each-study" class="section level2">
<h2 class="hasAnchor">
<a href="#pca-summary-for-each-study" class="anchor"></a>PCA summary for each study</h2>
<p>PCA summary of each study can be accessed through <code>PCAsummary</code> method. Output is a list with the length of studies, where each element is a matrix containing PCA summary results: standard deviation (SD), variance explained by each PC (Variance), and the cumulative variance explained (Cumulative).</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="../reference/PCAGenomicSignatures-methods.html">PCAsummary</a></span><span class="op">(</span><span class="va">RAVmodel</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] 536</span>
<span class="fu"><a href="../reference/PCAGenomicSignatures-methods.html">PCAsummary</a></span><span class="op">(</span><span class="va">RAVmodel</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>
<span class="co">#&gt; $DRP000987</span>
<span class="co">#&gt;            DRP000987.PC1 DRP000987.PC2 DRP000987.PC3 DRP000987.PC4</span>
<span class="co">#&gt; SD            42.6382804   21.97962707   12.94848367   11.89265640</span>
<span class="co">#&gt; Variance       0.3580557    0.09514629    0.03302091    0.02785537</span>
<span class="co">#&gt; Cumulative     0.3580557    0.45320196    0.48622287    0.51407824</span>
<span class="co">#&gt;            DRP000987.PC5 DRP000987.PC6 DRP000987.PC7 DRP000987.PC8</span>
<span class="co">#&gt; SD           10.89460616   10.53655583    9.47278967    8.97268098</span>
<span class="co">#&gt; Variance      0.02337622    0.02186495    0.01767287    0.01585607</span>
<span class="co">#&gt; Cumulative    0.53745446    0.55931942    0.57699228    0.59284836</span>
<span class="co">#&gt;            DRP000987.PC9 DRP000987.PC10 DRP000987.PC11 DRP000987.PC12</span>
<span class="co">#&gt; SD            8.43652010     7.90810771     7.60865968    7.076401846</span>
<span class="co">#&gt; Variance      0.01401774     0.01231676     0.01140165    0.009862254</span>
<span class="co">#&gt; Cumulative    0.60686609     0.61918285     0.63058449    0.640446748</span>
<span class="co">#&gt;            DRP000987.PC13 DRP000987.PC14 DRP000987.PC15 DRP000987.PC16</span>
<span class="co">#&gt; SD             6.89881271    6.766783788    6.703417100      6.6042884</span>
<span class="co">#&gt; Variance       0.00937346    0.009018116    0.008850009      0.0085902</span>
<span class="co">#&gt; Cumulative     0.64982021    0.658838324    0.667688332      0.6762785</span>
<span class="co">#&gt;            DRP000987.PC17 DRP000987.PC18 DRP000987.PC19 DRP000987.PC20</span>
<span class="co">#&gt; SD             6.44257163    6.245257583    6.042236748    5.992461574</span>
<span class="co">#&gt; Variance       0.00817466    0.007681604    0.007190294    0.007072317</span>
<span class="co">#&gt; Cumulative     0.68445319    0.692134796    0.699325091    0.706397408</span></code></pre></div>
</div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<details><div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.0.5 (2021-03-31)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 18.04.5 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3</span>
<span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] parallel  stats4    stats     graphics  grDevices utils     datasets </span>
<span class="co">#&gt; [8] methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] GenomicSuperSignature_1.1.2 SummarizedExperiment_1.20.0</span>
<span class="co">#&gt;  [3] Biobase_2.50.0              GenomicRanges_1.42.0       </span>
<span class="co">#&gt;  [5] GenomeInfoDb_1.26.7         IRanges_2.24.1             </span>
<span class="co">#&gt;  [7] S4Vectors_0.28.1            BiocGenerics_0.36.1        </span>
<span class="co">#&gt;  [9] MatrixGenerics_1.2.1        matrixStats_0.58.0         </span>
<span class="co">#&gt; [11] BiocStyle_2.18.1           </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] bitops_1.0-7           fs_1.5.0               bit64_4.0.5           </span>
<span class="co">#&gt;   [4] httr_1.4.2             RColorBrewer_1.1-2     rprojroot_2.0.2       </span>
<span class="co">#&gt;   [7] tools_4.0.5            backports_1.2.1        bslib_0.2.4           </span>
<span class="co">#&gt;  [10] utf8_1.2.1             R6_2.5.0               DBI_1.1.1             </span>
<span class="co">#&gt;  [13] colorspace_2.0-0       GetoptLong_1.0.5       withr_2.4.2           </span>
<span class="co">#&gt;  [16] tidyselect_1.1.0       bit_4.0.4              curl_4.3              </span>
<span class="co">#&gt;  [19] compiler_4.0.5         textshaping_0.3.3      Cairo_1.5-12.2        </span>
<span class="co">#&gt;  [22] desc_1.3.0             DelayedArray_0.16.3    bookdown_0.22         </span>
<span class="co">#&gt;  [25] sass_0.3.1             scales_1.1.1           rappdirs_0.3.3        </span>
<span class="co">#&gt;  [28] pkgdown_1.6.1          systemfonts_1.0.1      stringr_1.4.0         </span>
<span class="co">#&gt;  [31] digest_0.6.27          foreign_0.8-81         rmarkdown_2.7         </span>
<span class="co">#&gt;  [34] rio_0.5.26             XVector_0.30.0         pkgconfig_2.0.3       </span>
<span class="co">#&gt;  [37] htmltools_0.5.1.1      dbplyr_2.1.1           fastmap_1.1.0         </span>
<span class="co">#&gt;  [40] readxl_1.3.1           rlang_0.4.10           GlobalOptions_0.1.2   </span>
<span class="co">#&gt;  [43] RSQLite_2.2.7          shape_1.4.5            jquerylib_0.1.4       </span>
<span class="co">#&gt;  [46] generics_0.1.0         jsonlite_1.7.2         zip_2.1.1             </span>
<span class="co">#&gt;  [49] dplyr_1.0.5            car_3.0-10             RCurl_1.98-1.3        </span>
<span class="co">#&gt;  [52] magrittr_2.0.1         GenomeInfoDbData_1.2.4 Matrix_1.3-2          </span>
<span class="co">#&gt;  [55] Rcpp_1.0.6             munsell_0.5.0          fansi_0.4.2           </span>
<span class="co">#&gt;  [58] abind_1.4-5            lifecycle_1.0.0        stringi_1.5.3         </span>
<span class="co">#&gt;  [61] yaml_2.2.1             carData_3.0-4          zlibbioc_1.36.0       </span>
<span class="co">#&gt;  [64] BiocFileCache_1.14.0   blob_1.2.1             grid_4.0.5            </span>
<span class="co">#&gt;  [67] forcats_0.5.1          crayon_1.4.1           lattice_0.20-41       </span>
<span class="co">#&gt;  [70] haven_2.4.1            hms_1.0.0              circlize_0.4.12       </span>
<span class="co">#&gt;  [73] knitr_1.33             ComplexHeatmap_2.6.2   pillar_1.6.0          </span>
<span class="co">#&gt;  [76] ggpubr_0.4.0           rjson_0.2.20           ggsignif_0.6.1        </span>
<span class="co">#&gt;  [79] glue_1.4.2             evaluate_0.14          data.table_1.14.0     </span>
<span class="co">#&gt;  [82] BiocManager_1.30.12    png_0.1-7              vctrs_0.3.7           </span>
<span class="co">#&gt;  [85] cellranger_1.1.0       gtable_0.3.0           purrr_0.3.4           </span>
<span class="co">#&gt;  [88] tidyr_1.1.3            clue_0.3-59            assertthat_0.2.1      </span>
<span class="co">#&gt;  [91] cachem_1.0.4           ggplot2_3.3.3          openxlsx_4.2.3        </span>
<span class="co">#&gt;  [94] xfun_0.22              broom_0.7.6            rstatix_0.7.0         </span>
<span class="co">#&gt;  [97] ragg_1.1.2             tibble_3.1.1           memoise_2.0.0         </span>
<span class="co">#&gt; [100] cluster_2.1.1          ellipsis_0.3.1</span></code></pre></div>
</details>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Sehyun Oh, Levi Waldron, Sean Davis.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
